### PR TITLE
Init

### PR DESCRIPTION
**Pull request by sarkis**
## what
Initialize terraform module for AWS CodePipeline assets for ECS.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "s3" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.s3.arn}"
+}
+
+resource "aws_iam_policy" "s3" {
+  name   = "${module.label.id}-s3"
```
Comment (osterman) [inline main.tf]: The problem with taking this shortcut is we don't honor the `delimiter`. Should therefore create new label.

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,215 @@
+module "label" {
```
Comment (osterman) [inline main.tf]: Probably need a more specific name than `label`, since there will often be more than one label per module.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "codebuild" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.codebuild.arn}"
+}
+
+resource "aws_iam_policy" "codebuild" {
+  name   = "${module.label.id}-codebuild"
```
Comment (osterman) [inline main.tf]: Use new label.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+variable "build_image" {
+  default     = "aws/codebuild/docker:17.09.0"
+  description = "Docker image for build environment, _e.g._ `aws/codebuild/docker:docker:17.09.0`"
+}
+
+variable "build_compute_type" {
+  default     = "BUILD_GENERAL1_SMALL"
+  description = "`CodeBuild` instance size. Possible values are: ```BUILD_GENERAL1_SMALL``` ```BUILD_GENERAL1_MEDIUM``` ```BUILD_GENERAL1_LARGE```"
```
Comment (osterman) [inline variables.tf]: Let's remove the code blocks. Just use inline-code format.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,91 @@
-# terraform-aws-ecs-codepipeline
-Terraform Module for CI/CD with AWS Code Pipeline and Code Build for ECS https://cloudposse.com/
+![Cloud Posse](https://cloudposse.com/logo-300x69.png)
+
+# terraform-aws-ecs-codepipeline [![Build Status](https://travis-ci.org/cloudposse/terraform-aws-ecs-codepipeline.svg?branch=master)](https://travis-ci.org/cloudposse/terraform-aws-ecs-codepipeline)
+
+A Terraform module to generate well-formed JSON documents (container definitions) that are passed to the `aws_ecs_task_definition` Terraform resource.
```
Comment (osterman) [inline README.md]: Out-of-date

File: README.md [hunk 6]
```diff
@@ -1,2 +1,91 @@
...
+
+## Usage
+
+This module is meant to be used as output only, meaning it will be used to create outputs which are consumed as a parameter by Terraform resources or other modules.
```
Comment (osterman) [inline README.md]: out-of-date

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+    action {
+      name            = "Deploy"
+      category        = "Deploy"
+      owner           = "AWS"
+      provider        = "ECS"
+      input_artifacts = ["task"]
+      version         = "1"
+
+      configuration {
+        ClusterName = "${var.ecs_cluster_name}"
```
Comment (aknysh) [inline main.tf]: `var.ecs_cluster_name` not defined

File: main.tf (hunk 2 above)
Comment (aknysh) [reply main.tf]: `main.tf` needs `terraform fmt`

File: main.tf (hunk 7 above)
Comment (sarkis) [reply main.tf]: Added here if that's what you meant: https://github.com/cloudposse/terraform-aws-ecs-codepipeline/pull/1/files#diff-c9ac8098c5ea9d3e6a9a596ff0c512a4R21

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+
+resource "aws_iam_role_policy_attachment" "codebuild" {
+  role       = "${aws_iam_role.default.id}"
+  policy_arn = "${aws_iam_policy.codebuild.arn}"
+}
+
+
+module "codebuild_label" {
+  source     = "github.com/cloudposse/terraform-terraform-label.git?ref=0.1.2"
+  attributes = ["codebuild"]
```
Comment (osterman) [inline main.tf]: Add the incoming attributes to this as well.

Comment (osterman) [reply main.tf]: Like this... https://github.com/cloudposse/terraform-aws-ecs-codepipeline/pull/1/files#diff-7a370d8342e7203b805911c92454f0f4R166

**Pull request merged**